### PR TITLE
ref(guides): Refactor assistant guides frontend

### PR DESCRIPTION
**Pull request by maheskett**
Alright, round two of getting in these changes. This is a refactor of the frontend code around assistant guides, which are onboarding tooltips on the sidebar, issues and issue details page. The refactor relies on api changes that are behind a `v2` query param implemented in #17552. 

Changes:
- Converts GuideAnchor, GuideStore and actionCreators/guides to typescript
- Moves the guide tooltip content (title, description, etc.) to the frontend
- Removes integer guide ids in favor of guide string keys
- Removes jquery and a snapshot from guideAnchor

### CODE REVIEW

File: src/sentry/static/sentry/app/stores/guideStore.tsx [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+import {Client} from 'app/api';
+import {Guide, GuidesServerData, GuidesContent} from 'app/components/assistant/types';
+import {trackAnalyticsEvent} from 'app/utils/analytics';
+import ConfigStore from 'app/stores/configStore';
+import getGuidesContent from 'app/components/assistant/getGuidesContent';
+import GuideActions from 'app/actions/guideActions';
+import OrganizationsActions from 'app/actions/organizationsActions';
+
+const guidesContent: GuidesContent = getGuidesContent();
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/stores/guideStore.tsx]: Do you need to specify the type on this?

**Comment by EvanPurkhiser** [comment]
These are pretty much the same frontend changes from the previous PR right?

File: src/sentry/static/sentry/app/stores/guideStore.tsx (hunk 1 above)
Comment (maheskett) [reply src/sentry/static/sentry/app/stores/guideStore.tsx]: What's your thought? I think it adds typing here: https://github.com/getsentry/sentry/pull/17569/files#diff-cc70d94737d46a81ec2ea09a62bae355R103

**Comment by maheskett** [comment]
@EvanPurkhiser Yeah, it's pretty much the same except it no longer includes the backend changes, which have already been deployed

**Review by evanpurkhiser — APPROVED** [review]

**Review by billyvg — APPROVED** [review]

**Pull request merged**